app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TweetsController < ApplicationController
+   get '/tweets' do
+      logged_in? ? (erb :'tweets/tweets') : (redirect '/login')
+   end
+
+   get '/tweets/new' do
+      logged_in? ? (erb :'tweets/create_tweet') : (redirect '/login')
+   end
+
+   post '/tweets/new' do
+      if params[:content].empty?
+         redirect '/tweets/new'
+      else
+         Tweet.create(content: params[:content], user_id: current_user.id)
+         redirect '/tweets'
+      end
+   end
+
+   get '/tweets/:id' do
+      @tweet = Tweet.find_by(id: params[:id])
+      logged_in? ? (erb :'tweets/show_tweet') : (redirect '/login')
+   end
+
+   get '/tweets/:id/edit' do
+      if logged_in?
+         @tweet = Tweet.find_by(id: params[:id])
+         @tweet.user.id == current_user.id ? (erb :'tweets/edit_tweet') : (redirect '/tweets')
+      else
+         redirect '/login'
+      end
+   end
+
+   patch '/tweets/:id/edit' do
+      @tweet = Tweet.find_by(id: params[:id])
+      if params[:content].empty?
+         redirect "/tweets/#{@tweet.id}/edit"
+      else
+         @tweet.content = params[:content]
+         @tweet.save
+         redirect '/tweets'
+      end
+   end
+
+   delete '/tweets/:id/delete' do
+      @tweet = Tweet.find_by(id: params[:id])
+      if @tweet.user.id == current_user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.